LexicalAnalyzer.c

```diff
@@ -0,0 +1,488 @@
+// the string as a KEYWORD.
+// Returns the integer placeholder (i) to show how far into the line the string of letters ends.
+token *handleLetter (int* i, char* buffer, char* tempBuffer, token *head);
+
+// A separate function to handle characters that are not allowed including ! by itself.
+// Prints out ERROR: followed by the string of consecuative non-allowed characters.
+// Returns the integer placeholder (i) to show how far into the line the error string ends.
+int handleError(int i, char* buffer, char* tempBuffer);
+
+// This function allocates space for a token and initializes it with the
+// parameters passed to it. The newly initialized token is then returned.
+token *createToken(char *type, char *value, char *retType);
+
+// This function iterates to the end of the linked list of token and inserts a newly created
+// token initialized with the values of the parameters at the end. The head node is returned.
+token *insert(token *head, char *type, char *value, char *retType);
+
+// This function takes in the head node of the linked list and prints the type, value, and
+// depth of each element in the linked list. This is just for debugging.
+void print(token *head);
+
+void printHelp();
+
+
+
+int main (void) {
+	// declarations
+	char user_input[BUFSIZE];
+	char tempBuffer[BUFSIZE];
+	char subBuf[6];
+	char side[2];
+	int i;
+	int length;
+
+
+
+    // Initial screen
+    printf("                                                                      \n");
+    printf("::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::\n");
+    printf("::******************************************************************::\n");
+    printf("::**                                                              **::\n");
+    printf("::**                   SSSSSS       QQQQQQQ     LL                **::\n");
+    printf("::**                  SS     SS    QQ     QQ    LL                **::\n");
+    printf("::**                  SS          QQ       QQ   LL                **::\n");
+    printf("::**  ww       ww      SSSSSS     QQ       QQ   LL        xx  xx  **::\n");
+    printf("::**   w       w            SS    QQ     Q QQ   LL          xx    **::\n");
+    printf("::**    w  w  w      SS     SS     QQ     QQ    LL          xx    **::\n");
+    printf("::**     ww ww         SSSSSS       QQQQQQQ Q   LLLLLLL   xx  xx  **::\n");
+    printf("::**                                                              **::\n");
+    printf("::******************************************************************::\n");
+    printf("::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::\n");
+    printf("                                                                      \n");
+    printf("Welcome to wSQLx!\n");
+
+
+
+    // create the head node for the linked list of tokens
+    token *head = NULL;
+    token *tmp = NULL;
+
+    while (1) {  // Loop for terminal entries
```

Comment:
I don't want to leave a ton of feedback that isn't welcome, but this loop appears to be getting complex.

If you want I can pick it apart, but otherwise as long as it constructs a ParseTree struct, I don't mind!